src/amm.rs

```diff
@@ -45,25 +48,59 @@ macro_rules! to_u128 {
     };
 }
 
+macro_rules! multiply {
+    ($a:expr, $b:expr) => {
+        $a.checked_mul($b).expect(CANNOT_OVERFLOW)
+    };
+}
+
+macro_rules! add {
+    ($a:expr, $b:expr) => {
+        $a.checked_add($b).expect(CANNOT_OVERFLOW)
+    };
+}
+
+macro_rules! divide {
+    ($a:expr, $b:expr) => {
+        $a.checked_div($b).expect(DENOMINATOR_NOT_ZERO)
+    };
+}
+
+macro_rules! substract {
+    ($a:expr, $b:expr) => {
+        $a.checked_sub($b).expect(CANNOT_OVERFLOW)
+    };
+}
+
+
+#[derive(PartialEq)]
+#[derive(Debug)]
+pub enum MathError {
+    DivisorIsZero,
```

Comment:
I would go with something like : ZeroInReserve  ( renaming is part of another Pr but we can prepare already )